src/store/workflows.module.js

```diff
@@ -1,18 +1,104 @@
 // raw GraphQL response data structure
+import { extractGroupState } from '@/utils/tasks'
+
 const state = {
-  workflows: []
+  workflows: [],
+  workflowTree: []
 }
 
 const mutations = {
   SET (state, data) {
     // TODO: when subscriptions are introduced this will have to apply
     // deltas to the store
     state.workflows = data
+  },
+  SET_WORKFLOW_TREE (state, workflowTree) {
+    state.workflowTree = workflowTree
+  }
+}
+
+/**
+ * Compute a map of cycles, where the key is the workflow ID, and the
+ * value is the cyclepoint name.
+ *
+ * @param workflows
+ * @returns {Map<any, any>}
+ * @private
+ */
+function _getCycles (workflows) {
+  const cycles = new Map()
+  for (const workflow of workflows) {
+    if (Object.prototype.hasOwnProperty.call(workflow, 'taskProxies')) {
+      if (!cycles.get(workflow.id)) {
```

Comment:
Is `.id` some kind of JavaScript built-in? (It gets syntax highlighted in blue here.)

Reply:
Not that I am aware of. I think it is just the syntax highlight library getting confused.